firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -2915,6 +2918,7 @@ class BrowserViewController: UIViewController,
         case .newTab:
             willNavigateAway(from: tabManager.selectedTab)
             topTabsDidPressNewTab(tabManager.selectedTab?.isPrivate ?? false)
+            recordVisitManager.resetRecording()
```

Comment:
This could be tested too if we call `func newState(state: BrowserViewControllerState)` on BVC in a unit test?